sql/ddl_internal.sql

```diff
@@ -65,6 +66,22 @@ BEGIN
         USING ERRCODE ='IO101';
     END IF;
 
+    -- Pre-validate dimensions to ensure we do not unnecessarily
+    -- increment the hypertable ID
+    PERFORM _timescaledb_internal.validate_dimension(main_table,
```

Comment:
FWIW, I have another PR in the pipeline that will clean up `create_hypertable()`, which will make this stuff a lot nicer.